internal/notifier/factory.go

```diff
@@ -91,6 +92,8 @@ func (f Factory) Notifier(provider string) (Interface, error) {
 		n, err = NewAzureDevOps(f.ProviderUID, f.URL, f.Token, f.CertPool)
 	case apiv1.GoogleChatProvider:
 		n, err = NewGoogleChat(f.URL, f.ProxyURL)
+	case apiv1.GooglePubSubProvider:
+		n, err = NewGooglePubSubTopic(ctx, f.Channel, []byte(f.Token))
```

Comment:
I just saw this:

```go
			if webhook == "" {
				providerLogger.Error(nil, "provider has no address")
				continue
			}
```

And then realized that all the providers are using `f.URL`, which comes from the `webhook` variable above, which comes from the Provider `.spec.address` field (or `.data.address` from the referenced Secret when specified).

I used `.spec.channel` to encode both Google's project ID and topic ID using Google's resource name format, but I could also make it `.spec.address` the project ID and `.spec.channel` the topic ID. Would that make sense? Otherwise we'd need to remove this `if` for this new Provider type to work.

Reply:
Or a much better option: drop `channel` altogether and just use `address` instead with the current topic name regex

Reply:
Fixed!